### PR TITLE
Add hashicorp vault edc extension

### PR DESCRIPTION
**Pull request by bcronin90**
<sub>Brendan Cronin brendan.cronin@mercedes-benz.com, Mercedes Benz Tech Innovation GmbH, [legal info/Impressum](https://github.com/mercedes-benz/foss/blob/master/LEGAL_IMPRINT.md)</sub>

### CODE REVIEW

File: edc-extensions/hashicorp-vault/pom.xml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>2.0.0-alpha1</version>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>jul-to-slf4j</artifactId>
+            <version>2.0.0-alpha1</version>
+        </dependency>
```
Comment (florianrusch-zf) [inline edc-extensions/hashicorp-vault/pom.xml:104]: I think this are not test libraries or? If they are, the scope is missing. If they are not, then I would move it out of the test section

File: edc-extensions/hashicorp-vault/pom.xml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-all</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>junit-jupiter</artifactId>
+            <version>1.16.3</version>
+        </dependency>
```
Comment (florianrusch-zf) [inline edc-extensions/hashicorp-vault/pom.xml:82]: Is here the test scope missing?

File: edc-extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorp/CreateHashicorpVaultEntryRequestPayload.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  @Builder
+  @NoArgsConstructor
+  @AllArgsConstructor
+  @Data
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  static class Options {
+    @JsonProperty("cas")
+    private Integer cas;
+  }
```
Comment (florianrusch-zf) [inline edc-extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorp/CreateHashicorpVaultEntryRequestPayload.java:46]: Strange indentation? Don't we use 4 spaces?

File: edc-extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorp/CreateHashicorpVaultEntryRequestPayload.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+class CreateHashicorpVaultEntryRequestPayload {
```
Comment (florianrusch-zf) [inline edc-extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorp/CreateHashicorpVaultEntryRequestPayload.java:30]: I would recommend to rename the class to not have a verb in it. This should be the `HashicorpVaultEntryRequest` class.

```suggestion
class HashicorpVaultEntryRequest {
```

File: edc-extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorp/CreateHashicorpVaultEntryResponsePayload.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+class CreateHashicorpVaultEntryResponsePayload {
```
Comment (florianrusch-zf) [inline edc-extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorp/CreateHashicorpVaultEntryResponsePayload.java:29]:
```suggestion
class HashicorpVaultEntryResponse {
```

File: edc-extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorp/HashicorpVaultClient.java [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+                      future.completeExceptionally(
+                          new EdcException("Received an empty body response from vault"));
+                    } else {
+                      GetHashicorpVaultEntryResponsePayload payload =
+                          objectMapper.readValue(
+                              body.string(), GetHashicorpVaultEntryResponsePayload.class);
+                      String value =
+                          Objects.requireNonNull(payload.getData().getData().get(SECRET_KEY));
+                      future.complete(Result.success(value));
+                    }
```
Comment (florianrusch-zf) [inline edc-extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorp/HashicorpVaultClient.java:90]: You can get rid of this else if you return in the if body before:
```suggestion
                      return;
                    }
                    
                    GetHashicorpVaultEntryResponsePayload payload =
                        objectMapper.readValue(
                            body.string(), GetHashicorpVaultEntryResponsePayload.class);
                    String value =
                        Objects.requireNonNull(payload.getData().getData().get(SECRET_KEY));
                    future.complete(Result.success(value));
```

File: edc-extensions/hashicorp-vault/src/test/resources/logback.xml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<configuration>
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+        <encoder>
+            <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n</pattern>
+        </encoder>
+    </appender>
+    <root level="debug">
+        <appender-ref ref="STDOUT"/>
+    </root>
+</configuration>
```
Comment (florianrusch-zf) [inline edc-extensions/hashicorp-vault/src/test/resources/logback.xml:10]: Missing empty EOF

```suggestion
</configuration>

```

File: edc-extensions/pom.xml [hunk 8]
```diff
@@ -29,6 +29,7 @@
         <module>business-partner-validation</module>
         <module>aas-controller</module>
         <module>postgresql-migration</module>
+        <module>hashicorp-vault</module>
     </modules>
 
 </project>
```
Comment (florianrusch-zf) [inline edc-extensions/pom.xml:35]: Missing empty EOF

```suggestion
</project>

```

File: pom.xml [hunk 9]
```diff
@@ -791,6 +791,11 @@
                 <artifactId>web-spi</artifactId>
                 <version>${org.eclipse.dataspaceconnector.version}</version>
             </dependency>
+            <dependency>
+                <groupId>org.eclipse.dataspaceconnector</groupId>
+                <artifactId>junit-extension</artifactId>
+                <version>${org.eclipse.dataspaceconnector.version}</version>
+            </dependency>
```
Comment (florianrusch-zf) [inline pom.xml:798]: Test scope?
```suggestion
            <dependency>
                <groupId>org.eclipse.dataspaceconnector</groupId>
                <artifactId>junit-extension</artifactId>
                <version>${org.eclipse.dataspaceconnector.version}</version>
                <scope>test</scope>
            </dependency>
```

**Review by florianrusch-zf — CHANGES_REQUESTED** [review]
Hey, I know it's still a Draft PR, but I already take a view and saw a view things, which could be improved.

File: edc-extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorp/CreateHashicorpVaultEntryRequestPayload.java (hunk 3 above)
Comment (denisneuling) [reply edc-extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorp/CreateHashicorpVaultEntryRequestPayload.java:46]: Atm we use [Google Java Style](https://google.github.io/styleguide/javaguide.html) resp. [spotless google-java-format](https://github.com/diffplug/spotless/tree/main/plugin-maven#google-java-format). But let's discuss using [Eclipse JDT](https://github.com/diffplug/spotless/tree/main/plugin-maven#eclipse-jdt) to be better aligned with the upstream EDC repo.

<sub>Denis Neuling <denis.neuling@mercedes-benz.com>, Mercedes-Benz Tech Innovation GmbH, [legal info/Impressum](https://github.com/mercedes-benz/daimler-foss/blob/master/LEGAL_IMPRINT.md) </sub>

File: edc-extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorp/CreateHashicorpVaultEntryRequestPayload.java (hunk 4 above)
Comment (denisneuling) [reply edc-extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorp/CreateHashicorpVaultEntryRequestPayload.java:30]: The hashicorp vault API expects different payloads for creating/updating keys, also it's not responding according to that structure, so I guess it's better to express explicitly what this class is used for. Furthermore IMHO `Request` indicates the object contains all details to construct a HTTP request object - including headers, uri, method etc. - which is not the case so I'd prefer keeping that within name. 


<sub>Denis Neuling <denis.neuling@mercedes-benz.com>, Mercedes-Benz Tech Innovation GmbH, [legal info/Impressum](https://github.com/mercedes-benz/daimler-foss/blob/master/LEGAL_IMPRINT.md) </sub>

Comment (florianrusch-zf) [reply edc-extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorp/CreateHashicorpVaultEntryRequestPayload.java:30]: Ahhhh got it! Then I'm totally with you ;-) Thanks for explaining this!

File: edc-extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorp/CreateHashicorpVaultEntryRequestPayload.java (hunk 3 above)
Comment (florianrusch-zf) [reply edc-extensions/hashicorp-vault/src/main/java/org/eclipse/dataspaceconnector/core/security/hashicorp/CreateHashicorpVaultEntryRequestPayload.java:46]: Then we will let it as it is and discuss this team internally 😉

File: edc-extensions/hashicorp-vault/pom.xml (hunk 1 above)
Comment (denisneuling) [reply edc-extensions/hashicorp-vault/pom.xml:104]: Yeah, looks strange, but the scope here needs to be `test`... These dependencies are just used to see the test-containers logs. the extension itself uses the edc monitor and therefore is agnostic towards which logging subsystem is used.

**Comment by bcronin90** [comment]
@denisneuling Just lost Internet access. Using UMTS for now, but should be fixed by tomorrow. Can't use teams because it's too slow, but I'll get the remaining unit tests up to where we left the current one.

**Comment by denisneuling** [comment]
@bcronin90 Shall we close this one?

**Comment by bcronin90** [comment]
> @bcronin90 Shall we close this one?

I think we can. Unless we want cert authentication any time soon. Which I'm not sure is required or worth it.

**Pull request closed without merging**